c++/src/kj/compat/http.c++

```diff
@@ -4968,21 +4947,22 @@ private:
     currentMethod = nullptr;
```

Comment:
It appears that every branch below now either re-assigns currentMethod back to its original value, except the one branch that sets it to GET. Maybe remove this line and all the `currentMethod = method` lines below.

Reply:
Maybe we should `currentMethod = HttpMethod::GET` here, actually? That would allow us to return an error response correctly if someone tried to open a WebSocket connection with a HEAD request (admittedly unlikely).

Reply:
The very first check below this line verifies that the method is GET, so I think that's redundant. (In fact I guess this means that *all* of the later assignments to `currentMethod` can be deleted if we remove this line that sets it null.)

Reply:
(And note that it would be incorrect to respond to a HEAD request with a bodied response even if the HEAD request is an invalid WebSocket handshake. So no, we should not change HEAD to GET.)

Reply:
Got it!